misk-aws/src/main/kotlin/misk/jobqueue/sqs/AwsSqsJobQueueModule.kt

```diff
@@ -110,30 +113,62 @@ class AwsSqsJobQueueModule(
+      config: AwsSqsJobQueueConfig,
+      credentials: AWSCredentialsProvider,
+      region: AwsRegion,
+      forSqsReceiving: Boolean
     ): AmazonSQS {
-      // Defaults from SDK
-      val clientConfiguration = ClientConfiguration()
-          .withSocketTimeout(25000)
-          .withConnectionTimeout(1000)
 
-      val builder = AmazonSQSClientBuilder.standard()
+      return if (forSqsReceiving) {
+        // We don't need any buffering functionality for receiving; build a regular sync client.
+        AmazonSQSClientBuilder.standard()
           .withCredentials(credentials)
           .withRegion(region.name)
-      if (forSqsReceiving) {
-        // Do not artificially constrain the # of connections to SQS. Instead we rely on higher
-        // level resource limiting knobs (e.g # of parallel receivers).
-        // We only do this for receiving, as sending does not have equivalent knobs.
-        builder.withClientConfiguration(clientConfiguration.withMaxConnections(Int.MAX_VALUE))
+          .withClientConfiguration(ClientConfiguration()
+            .withSocketTimeout(25_000)
+            .withConnectionTimeout(1_000)
+            // Do not artificially constrain the # of connections to SQS. Instead we rely on higher
+            // level resource limiting knobs (e.g # of parallel receivers).
+            // We only do this for receiving, as sending does not have equivalent knobs.
+            .withMaxConnections(Int.MAX_VALUE))
+          .build()
       } else {
-        builder.withClientConfiguration(clientConfiguration
+        // Use a buffered client for sending, to group enqueues and deletes into batches.
+        // The trade-off is fewer network calls (which costs less $ since SQS cost is directly
+        // proportional to API calls) at a slightly higher individual call duration.
+        // Every call individual SendMessage and DeleteMessage request will wait up to 200ms
+        // for similar requests and sent in a batch. Batches are immediately sent once full.
+        val asyncClient = AmazonSQSAsyncClientBuilder.standard()
+          .withCredentials(credentials)
+          .withRegion(region.name)
+          .withClientConfiguration(ClientConfiguration()
             .withSocketTimeout(config.sqs_sending_socket_timeout_ms)
             .withConnectionTimeout(config.sqs_sending_connect_timeout_ms)
-            .withRequestTimeout(config.sqs_sending_request_timeout_ms)
-        )
+            .withRequestTimeout(config.sqs_sending_request_timeout_ms))
+          .build()
+
+        // NB: It's unlikely this client will be used for receiving (i.e. to issue ReceiveMessage
+        // requests) but turn off pre-fetching in any case. Without pre-fetching, receives are
+        // on-demand (i.e. work exactly the same as a non-buffered client) and won't spawn extra
+        // background threads to pre-fill receive buffers.
+        val bufferConfig = QueueBufferConfig().withNoPrefetching()
```

Comment:
One of the options of `QueueBufferConfig` we may want to increase is `maxInflightOutboundBatches` from its default of 5 — meaning a max of 50 send/delete requests being sent to SQS per queue, at any given instant.